Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine+State.swift

```diff
@@ -193,11 +193,15 @@ extension ExperienceStateMachine.Transition {
                 sideEffect: .error(.step(experience, currentIndex, "Step at \(stepRef) does not exist"), reset: false))
         }
 
+        let stepGroup = experience.steps[stepIndex.group]
+        let navigationActions = stepGroup.actions[stepGroup.id.uuidString.lowercased()]?.filter { $0.trigger == "navigate" } ?? []
```

Comment:
Unrelated to the main idea here, but this reminds me: we've done the `.lowercased()` thing enough times that we should have an extension on the UUID type that gets a lowercased "appcues compatible" version of a UUID string.